modules/bootstrap-launcher/src/it/scala/coursier/bootstrap/launcher/DownloadTests.scala

```diff
@@ -0,0 +1,96 @@
+package coursier.bootstrap.launcher
+
+import utest._
+
+import coursier.bootstrap.launcher.credentials.Credentials
+import coursier.bootstrap.launcher.credentials.DirectCredentials
+import coursier.paths.CachePath
+
+import scala.jdk.CollectionConverters._
+import java.nio.file.Files
+import java.io.File
+import java.nio.file.Path
+import java.net.URL
+import java.nio.file.Paths
+import java.net.URI
+import java.util.Collections
+
+object DownloadTests extends TestSuite {
+  
+  private val testRepository = Option(System.getenv("TEST_REPOSITORY")).getOrElse(sys.error("TEST_REPOSITORY not set"))
+  private val testRepositoryUser = Option(System.getenv("TEST_REPOSITORY_USER")).getOrElse(sys.error("TEST_REPOSITORY_USER not set"))
+  private val testRepositoryPassword = Option(System.getenv("TEST_REPOSITORY_PASSWORD")).getOrElse(sys.error("TEST_REPOSITORY_PASSWORD not set"))
+  private val testRepositoryHost = new URI(testRepository).getHost
+
+  private def deleteRecursive(f: File): Unit = {
+    if (f.isDirectory)
+      f.listFiles().foreach(deleteRecursive)
+    if (f.exists())
+      f.delete()
+  }
+
+  private def withTmpDir[T](f: Path => T): T = {
+    val dir = Files.createTempDirectory("coursier-test")
+    val shutdownHook: Thread =
+      new Thread {
+        override def run() =
+          deleteRecursive(dir.toFile)
+      }
+    Runtime.getRuntime.addShutdownHook(shutdownHook)
+    try f(dir)
+    finally {
+      deleteRecursive(dir.toFile)
+      Runtime.getRuntime.removeShutdownHook(shutdownHook)
+    }
+  }
+
+  val tests = Tests {
+
+    test("public URL") {
+      withTmpDir { dir =>
+        val remoteUrls = Seq(
+          new URL("https://repo1.maven.org/maven2/com/chuusai/shapeless_2.12/2.3.3/shapeless_2.12-2.3.3.jar")
+        )
+        val download = new Download(1, CachePath.defaultCacheDirectory(), Collections.emptyList())
```

Comment:
Shouldn't `dir` be used here?
```suggestion
        val download = new Download(1, dir, Collections.emptyList())
```

Reply:
Thanks for double checking. Yes, `CachePath.defaultCacheDirectory()` should have been `dir.toFile`. PR updated and checks out.